wallet/src/de/schildbach/wallet/service/UpgradeWalletService.java

```diff
@@ -38,7 +43,11 @@
  */
 public final class UpgradeWalletService extends IntentService {
     public static void startUpgrade(final Context context) {
-        context.startService(new Intent(context, UpgradeWalletService.class));
+        final Intent intent = new Intent(context, UpgradeWalletService.class);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
```

Comment:
This if block can also be replaced with ContextCompat.startForegroundService(context, intent).